rest/rest-api/src/main/java/org/ow2/proactive_grid_cloud_portal/scheduler/client/SchedulerRestClient.java

```diff
@@ -429,20 +434,29 @@ public ListFile list(String sessionId, String dataspacePath, String pathname) th
         }
     }
 
-    private JobIdData submit(String sessionId, InputStream job, MediaType mediaType, Map<String, String> variables)
-            throws Exception {
+    private JobIdData submit(String sessionId, InputStream job, MediaType mediaType, Map<String, String> variables,
+            Map<String, String> genericInfos) throws Exception {
         String uriTmpl = restEndpointURL + addSlashIfMissing(restEndpointURL) + "scheduler/submit";
 
         ResteasyClient client = new ResteasyClientBuilder().httpEngine(httpEngine)
                                                            .providerFactory(providerFactory)
                                                            .build();
         ResteasyWebTarget target = client.target(uriTmpl);
+
+        // Variables
         if (variables != null) {
             for (String key : variables.keySet()) {
                 target = target.matrixParam(key, variables.get(key));
             }
         }
 
+        // Generic infos
+        if (genericInfos != null) {
+            for (String key : genericInfos.keySet()) {
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Iterate over the "entrySet" instead of the "keySet". [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS2864)

Reply:
Not in our case